src/app/shared/components/guided-tour/directive.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {AfterViewInit, Directive, Input, OnDestroy, ViewContainerRef, ElementRef, HostListener} from '@angular/core';
+import {Subject} from 'rxjs';
+import {GuidedTourStep} from './entity';
+import {DialogService} from './dialog/service';
+import {DialogHelperService} from './dialog/helper.service';
+
+@Directive({
+  selector: 'kmGuidedTour, [kmGuidedTour]',
+})
+export class GuidedTourDirective implements AfterViewInit, OnDestroy {
+  @Input('kmGuidedTour') id: string;
+  @Input('kmGuidedTourWithBackdrop') withBackdrop = true;
+  @Input('kmGuidedTourIDToListen') idToListen: string;
+  @Input('kmGuidedTourPosition') position = 'bottom';
+
+  @HostListener('click', ['$event'])
+  onClick(event) {
+    if (
+      !!this._dialogHelperService.isTourInProgress() &&
+      (event.target.id === this.idToListen || event.target.parentElement.id === this.idToListen)
```

Comment:
If I understand this correctly, this `idToListen` seems like a boolean that checks if clicking this element triggers moving to the next step. It determines if clicking an element should trigger the next step. We probably will need to discuss the functionality again.

Reply:
more or less a boolean, but yes should be checking if the clicked event was the one for the next step